packages/lightning-lsp-common/src/fs-utils.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import * as fs from 'fs';
+
+/**
+ * Check if a file or directory exists asynchronously
+ */
+export const pathExists = async (filePath: string): Promise<boolean> => {
+    try {
+        await fs.promises.access(filePath);
+        return true;
+    } catch {
+        return false;
+    }
+};
+
+/**
+ * Ensure a directory exists, creating it if necessary (async)
+ */
+export const ensureDir = async (dirPath: string): Promise<void> => {
+    if (!fs.existsSync(dirPath)) {
+        fs.mkdirSync(dirPath, { recursive: true });
+    }
+};
+
+/**
+ * Ensure a directory exists, creating it if necessary (sync)
+ */
+export const ensureDirSync = (dirPath: string): void => {
+    if (!fs.existsSync(dirPath)) {
+        fs.mkdirSync(dirPath, { recursive: true });
+    }
+};
+
+/**
+ * Remove a file if it exists (safe file removal)
+ */
+export const removeFile = (filePath: string): void => {
```

Comment:
fs.rm with `force:true` will rm, if it exists.  https://nodejs.org/api/fs.html#fsrmpath-options-callback